backend/fms_core/tests/test_template_importers/library_preparation_test.py

```diff
@@ -0,0 +1,222 @@
+                                              execution_date=self.library_2['date'],
+                                              volume_used=self.library_2['volume_used'])
+        self.assertEqual(pm_2.volume_used, self.library_2['volume_used'])
+        self.assertEqual(pm_2.comment, self.library_2['library_comment'])
+        self.assertEqual(pm_2.process.protocol.name, 'Library Preparation')
+        self.assertEqual(pm_2.process.comment, self.library_2['batch_comment'])
+
+        # Test if pm_2' process and pm_1's process are the same since they come from the same batch
+        self.assertEqual(pm_1.process, pm_2.process)
+        
+        # Test third library
+        sample_library_3, _, _ = get_sample_from_container(barcode='Container4Library3')
+        library_3 = sample_library_3.derived_sample_not_pool.library
+
+        self.assertEqual(sample_library_3.volume, self.library_3['library_volume'])
+
+        # Library info tests
+        self.assertEqual(library_3.library_type, self.library_3['library_type'])
+        self.assertEqual(library_3.platform, self.library_3['platform'])
+        self.assertEqual(library_3.index, self.library_3['index'])
+        self.assertEqual(library_3.strandedness, self.library_3['strandedness'])
+
+        # Process and process measurements tests
+        pm_3 = ProcessMeasurement.objects.get(source_sample=source_sample,
+                                              execution_date=self.library_3['date'],
+                                              volume_used=self.library_3['volume_used'])
+        self.assertEqual(pm_3.volume_used, self.library_3['volume_used'])
+        self.assertEqual(pm_3.comment, self.library_3['library_comment'])
+        self.assertEqual(pm_3.process.protocol.name, 'Library Preparation')
+        self.assertEqual(pm_3.process.comment, self.library_3['batch_comment'])
+
+        # Property Values tests
+        pt_1 = PropertyType.objects.get(name='Library Technician Name', object_id=pm_3.process.protocol.id)
+        p_1 = PropertyValue.objects.get(property_type_id=pt_1, object_id=pm_3.process.id)
+
+        pt_2 = PropertyType.objects.get(name='Library Kit Used', object_id=pm_3.process.protocol.id)
+        p_2 = PropertyValue.objects.get(property_type_id=pt_2, object_id=pm_3.process.id)
+
+        pt_3 = PropertyType.objects.get(name='Library Kit Lot', object_id=pm_3.process.protocol.id)
+        p_3 = PropertyValue.objects.get(property_type_id=pt_3, object_id=pm_3.process.id)
+
+        pt_4 = PropertyType.objects.get(name='Shearing Technician Name', object_id=pm_3.process.protocol.id)
+        p_4 = PropertyValue.objects.get(property_type_id=pt_4, object_id=pm_3.process.id)
+
+        self.assertEqual(p_1.value, self.library_3['library_technician_name'])
+        self.assertEqual(p_2.value, self.library_3['library_kit_used'])
+        self.assertEqual(p_3.value, self.library_3['library_kit_lot'])
+        self.assertEqual(p_4.value, ' ')
```

Comment:
Unset properties are equal to one space character ? This is not ideal ...

Reply:
I agree. When the importer was being developed I suggested that empty property values were not created, but it was the way it was set.